composables/useCollectionAttributes.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery } from '@tanstack/vue-query'
+import type { NFT, Attribute } from '@/types'
+
+export const useCollectionAttributes = (collectionId: ComputedRef<string | undefined>) => {
+  const { data: nftsListData } = useQuery<NFT[]>({
+    queryKey: ['nft-attributes-by-collection', collectionId],
+    queryFn: async () =>
+      (collectionId.value
+        ? (await useAsyncGraphql<{ nfts: NFT[] } >({
+            query: 'nftAttributesListByCollection',
+            variables: { id: collectionId.value },
+          })).data.value.nfts
+        : []),
```

Comment:
>I think we should use useQuery from vue-apollo over the useGraphql  composable

I meant `useQuery` from vue-apollo, which does the same as `useGraphql`

![CleanShot 2025-01-18 at 09 05 11@2x](https://github.com/user-attachments/assets/135678b4-f673-4d2a-9c94-aa942b9bc0da)

`tanstack/vue-query` **can't be used outside setup scope**, it does not matter in this case but still something to keep in mind

Reply:
I would use `useQuery` from `tanstack/vue-query` here as it handles the reactive of variables and `refetching` better than `useGraphql`.